lib/mongo/uri/options_mapper.rb

```diff
@@ -472,62 +513,82 @@ def convert_auth_mech_props(name, value)
         properties
       end
 
+      # Reverts auth mechanism properties.
+      #
+      # @param [ Hash | nil ] value The auth mech properties.
+      #
+      # @return [ String | nil ] The auth mechanism properties as a string.
       def revert_auth_mech_props(value)
-        value
+        return if value.nil?
+        value.map { |k, v| "#{k}:#{v}" }.join(',')
       end
 
       # Parses the max staleness value, which must be either "0" or an integer
       # greater or equal to 90.
       #
       # @param [ String ] name Name of the URI option being processed.
-      # @param value [ String ] The max staleness string.
+      # @param [ String | Integer ] value The max staleness string.
       #
       # @return [ Integer | nil ] The max staleness integer parsed out if it is valid, otherwise nil
       #   (and a warning will be logged).
       def convert_max_staleness(name, value)
-        if /\A-?\d+\z/ =~ value
-          int = value.to_i
+        int = if value.is_a?(String) && /\A-?\d+\z/ =~ value
+          value.to_i
+        elsif value.is_a?(Integer)
+          value
+        end
 
-          if int == -1
-            int = nil
-          end
+        if int.nil?
```

Comment:
When `value` is neither string nor integer this branch won't be taken and code will continue to the comparison with -1 which looks to be wrong.

Reply:
I don't think that's correct... If it's neither a string or an integer, int will be nil and then warn and exit before it gets to that point... I should just combine those branches, however

Reply:
I see, sneaky.